src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -111,9 +154,18 @@ public InetSocketAddress next(long spinDelay) {
             lastIndex = 0;
         }
 
-        return serverAddresses.get(currentIndex);
+        InetSocketAddress curAddr = serverAddresses.get(currentIndex);
+
+        String curHostString = getHostString(curAddr);
+        List<InetAddress> resolvedAddresses = new ArrayList<InetAddress>(Arrays.asList(this.resolver.getAllByName(curHostString)));
+        if (resolvedAddresses.isEmpty()) {
+            throw new UnknownHostException("No IP address returned for address: " + curHostString);
```

Comment:
Should we try with the next on the list rather than throwing? After all, the call is to get the "next" resolved address. Of course, it is possible that we go through the whole list an find nothing. We might want to throw then so that we don't loop forever.

Reply:
I believe the expectation - and the original behaviour- is to loop forever until the DNS problem is fixed. Even if we don't do it here, the caller will.

I'd loop here infinitely. Also should keep an eye on spinDelays to avoid hammering the logic.

Reply:
I think it is fine to loop indefinitely because the client needs it for progress. We need a way to break the loop in the case the client closes, though.

Reply:
> We need a way to break the loop in the case the client closes, though.

That's actually a good reason for _not_ dealing with the error here. Because the caller - ClientCnxn - is be able to detect client closes, but StatisHostProvider is not.

Reply:
Ok.